test/e2e/collector/eraser_test.go

```diff
@@ -103,12 +123,110 @@ func TestRemoveImagesFromAllNodes(t *testing.T) {
+			if err != nil {
+				t.Error("Unable to uninstall deployment for teardown", err)
+			}
+			return ctx
+		}).
+		Feature()
+
+	disableScanFeat := features.New("Test Scanner Disabled Prune").
+		Setup(func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			wd, err := os.Getwd()
+			if err != nil {
+				t.Error("Could not get working directory", err)
+			}
+
+			providerResourceAbsolutePath, err := filepath.Abs(filepath.Join(wd, "/../../../", providerResourceDirectory, "eraser"))
+			if err != nil {
+				t.Error("Unable to get provider resource absolute path", err)
+			}
+
+			err = util.HelmInstall(cfg.KubeconfigFile(), "eraser-system", []string{providerResourceAbsolutePath, "--set", "scanner.image.repository="})
+			if err != nil {
+				t.Error("Unable to install deployment with scanner disabled", err)
+			}
+
+			return ctx
+		}).
+		Assess("ImageCollector CR is generated", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			c, err := cfg.NewClient()
+			if err != nil {
+				t.Error("Failed to create new client", err)
+			}
+
+			imagecollector := eraserv1alpha1.ImageCollector{}
+			wait.For(func() (bool, error) {
+				err := c.Resources().Get(ctx, "imagecollector-shared", "default", &imagecollector)
+				if err != nil {
+					t.Error("Could not get imagecollector-shared")
+				}
+
+				if imagecollector.ObjectMeta.Name == "imagecollector-shared" {
+					return true, nil
+				}
+
+				return false, nil
+			}, wait.WithTimeout(time.Minute*3))
+
+			return ctx
+		}).
+		Assess("ImageList Spec Contains Same Images As ImageCollecotor Shared", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			c, err := cfg.NewClient()
+			if err != nil {
+				t.Error("Failed to create new client", err)
+			}
+
+			// verify imagelist created
+			imagelist := eraserv1alpha1.ImageList{}
+			wait.For(func() (bool, error) {
+				err := c.Resources().Get(ctx, "imagelist", "default", &imagelist)
+				if util.IsNotFound(err) {
+					return false, nil
+				}
+
+				if err != nil {
+					return false, err
+				}
+
+				if imagelist.ObjectMeta.Name == "imagelist" {
+					return true, nil
+				}
+
+				return false, nil
+			}, wait.WithTimeout(time.Minute*3))
+
+			imagecollectorShared := eraserv1alpha1.ImageCollector{}
+			err = c.Resources().Get(ctx, "imagecollector-shared", "default", &imagecollectorShared)
+			if err != nil {
+				t.Error("Could not get imagecollector-shared")
+			}
+
+			// verify imagecollector-shared status fields are empty
+			if imagecollectorShared.Status.Vulnerable != nil || imagecollectorShared.Status.Failed != nil {
+				t.Error("Scan job has run, should be disabled")
+			}
+
+			// verify the images in both lists match
+			for _, img := range imagecollectorShared.Spec.Images {
+				// we add to imagelist by digest when prune without scanner
+				if !util.Contains(imagelist.Spec.Images, img.Digest) {
```

Comment:
Make a `map[string]struct{}` here for `imagecollectorShared.Spec.Images` to check for set membership; the nested linear search is `O(n²)`. Probably doesn't matter now, but since we copy/paste these tests it could get out of hand as the number of tests grow.